src/thunderbots/software/ai/primitive/catch_primative.h

```diff
@@ -0,0 +1,44 @@
+#pragma once
+
+#include <geom/angle.h>
+#include "primitive.h"
+
+class CatchPrimitive : public Primitive
+{
+public:
+    static const std::string PRIMITIVE_NAME;
+
+    /**
+     *
+     * Creates a new Catch Primative
+     *
+     * @param robot_id The id of the Robot to run this Primitive
+     * @param velocity Velocity to move robot forwards/backwards
+     * to catch the ball without it bouncing off the dribbler; units are m/s
+     * @param dribbler_speed Speed to rotate dribbler at. Units are RPM.
+     * @param margin THe margin of allowed error in the robot catch velocity.
+     * The exact use of this parameter needs to be confirmed.
```

Comment:
@MathewMacDougall - can we confirm what this is used (or not used) for?

@matthewberends - if we're not using this, we should be able to just remove it (we can always add it back later)

Reply:
Confirmed it is used, and is a scaling factor for how far in front of the ball to make the point of intercept. It scales a value based on some kind of difference in velocity between the ball and the robot (not immediately obvious how/why that relative velocity was chosen)

Eg. When the robot is in the position to catch the ball (on the ball's line of travel, in front of it, facing the ball), the ball should be `margin * magic number` meters away.